packages/react/src/components/Popup/Popup.tsx

```diff
@@ -62,6 +62,9 @@ export interface PopupProps
   /** Alignment for the popup. */
   align?: Alignment
 
+  /** If this prop is used, the Popup will propagate the key down event, otherwise it won't. */
+  allowKeyDownPropagation?: boolean
```

Comment:
See my comment: https://github.com/stardust-ui/react/pull/987/files#r263011991 No better proposal for now..